### PR TITLE
GUVNOR-2975: Guided Decision Table Graph: When graph is deleted the new one can't be created

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/GUVNOR-2975

https://issues.jboss.org/browse/GUVNOR-2977 was also caused by the same issue.

### CODE REVIEW

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableGraphEditorServiceImpl.java [hunk 1]
```diff
@@ -104,213 +107,205 @@ public GuidedDecisionTableGraphEditorServiceImpl( final @Named("ioStrategy") IOS
...
         final List<Path> paths = new ArrayList<>();
-        final DirectoryStream<org.uberfire.java.nio.file.Path> directoryStream = ioService.newDirectoryStream( nioRootPath );
-        for ( org.uberfire.java.nio.file.Path nioPath : directoryStream ) {
-            final Path path = Paths.convert( nioPath );
-            if ( resourceType.accept( path ) ) {
-                paths.add( path );
+        final DirectoryStream<org.uberfire.java.nio.file.Path> directoryStream = ioService.newDirectoryStream(nioRootPath);
+        for (org.uberfire.java.nio.file.Path nioPath : directoryStream) {
+            final Path path = Paths.convert(nioPath);
+            if (!dotFileFilter.accept(nioPath) && resourceType.accept(path)) {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/main/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableGraphEditorServiceImpl.java]: This is the fix.. to also filter "dot files" (there were other approaches, but using ```DotFileFilter``` made testing much easier).

File: drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/test/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableGraphEditorServiceImplTest.java [hunk 2]
```diff
@@ -114,199 +115,219 @@
...
+                                                                   Collectors.toSet())));
+        assertTrue(fileNames.contains("dtable1.gdst"));
+        assertTrue(fileNames.contains("dtable2.gdst"));
+        assertTrue(fileNames.contains("dtable3.gdst"));
     }
 
-    private org.uberfire.java.nio.file.Path makeNioPath( final String uri ) {
-        return Paths.get( uri );
+    @Test
+    public void testListDecisionTablesInPackageExcludesDotFiles() {
```
Comment (manstis) [inline drools-wb-screens/drools-wb-guided-dtable-editor/drools-wb-guided-dtable-editor-backend/src/test/java/org/drools/workbench/screens/guided/dtable/backend/server/GuidedDecisionTableGraphEditorServiceImplTest.java]: New test.

**Comment by manstis** [comment]
@Rikkola @karreiro Could you please review?

**Comment by manstis** [comment]
@etirelli Please advise if you want this in 7.0.0.Final (I assume "yes", but we need approval).

@jomarko I assume you'd want these fixes in 7.0.0 (if you Edson needs convincing).

**Comment by etirelli** [comment]
+1, yes, please, backport to 7.0.0.

**Pull request merged**